spec/acceptance/version_changer_spec.rb

```diff
@@ -30,5 +30,45 @@
       its(:stderr) { is_expected.to match(%r{Using Puppet file://}i) }
       its(:exit_status) { is_expected.to eq(0) }
     end
+
+    describe 'should use the correct puppet env' do
+      before(:all) do
+        File.open(File.join('manifests', 'init.pp'), 'w') do |f|
+          f.puts <<-PPFILE
+# version_select
+class version_select {
+}
+          PPFILE
+        end
+
+        FileUtils.mkdir_p(File.join('spec', 'classes'))
+        File.open(File.join('spec', 'classes', 'version_select_spec.rb'), 'w') do |f|
+          f.puts <<-TESTFILE
+require 'spec_helper'
+
+describe 'version_select', :if => Gem::Specification.find_by_name('puppet').source.options.key?('path') do
+  on_supported_os.each do |os, os_facts|
+    context 'on #{os}' do
+      let(:facts) { os_facts }
+
+      it { is_expected.to compile }
+    end
+  end
+end
+          TESTFILE
```

Comment:
This method seems a little round-about to me...

Could we update the test manifest to have two specific tests that assert either the presence or absence of the `path` key and then below use `pdk test unit --tests=` to only invoke the specific test and assert that it passed?

Reply:
I've updated the test to expect the path... so the acceptance test was simplified to expect a PDK test success or failure.